mix.exs

```diff
@@ -39,7 +39,7 @@ defmodule EXGBoost.MixProject do
     [
       {:elixir_make, "~> 0.4", runtime: false},
       {:nimble_options, "~> 1.0"},
-      {:nx, "~> 0.5"},
+      {:nx, github: "elixir-nx/nx", sparse: "nx"},
```

Comment:
Is this needed? I don't think we can push to hex with a git dependency. Otherwise, it looks good to go. I think a lot of the documentation changes are similar to the ones that @polvalente made in his PR as well. Thanks for the PR